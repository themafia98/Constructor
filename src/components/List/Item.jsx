import React, {useState} from 'react';

function Item(props) {
    const [id] = useState(props.id);
    const [item] = useState({...props.item});
    return (
        <div className = 'item' data-id = {id}>{item.name}</div>
    )
}

export default Item;