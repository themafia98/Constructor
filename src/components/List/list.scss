.projectsList__list {

    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    width: 80vw;

    display: grid;
    grid-template-columns: repeat(4, 20vw);
    text-align: center;

    .item {
        flex: 0 0 20vw;
        cursor: pointer;
        height: 20vh;
        border: 1px solid red;
        margin: 5px;
    }
}


@media screen and (max-width: 850px){

    .projectsList__list {

        grid-template-columns: repeat(3, 25vw);
    }
    .item {
        flex: 0 0 25vw;
    }
}

@media screen and (max-width: 650px){

    .projectsList__list {
        width: 100%;
        grid-template-columns: repeat(2, 1fr);
    }

    .item {
        flex: 0 0 50%;
    }
}


@media screen and (max-width: 450px){

    .projectsList__list {
        width: 100%;
        grid-template-columns: 1fr;
    }

    .item {
        flex: 0 0 100%;
    }
}