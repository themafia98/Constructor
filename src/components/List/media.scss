@media screen and (max-width: 850px){

    .projectsList__list {

        grid-template-columns: repeat(3, 25vw);
    }
    .item {
        flex: 0 0 25vw;
    }
}

@media screen and (max-width: 650px){

    .projectsList__list {
        width: 100%;
        grid-template-columns: repeat(2, 1fr);
    }

    .item {
        flex: 0 0 50%;
    }
}


@media screen and (max-width: 450px){


    .projectsList__list {
        width: 100%;
        grid-template-columns: 1fr;
    }

    .item {
        flex: 0 0 100%;
    }
}