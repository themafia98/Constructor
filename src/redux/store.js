import {createStore} from 'redux';
import combineReducers from './reducers';

let broswerPluginDev = window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__();
const store = createStore(combineReducers,broswerPluginDev);
export default store;