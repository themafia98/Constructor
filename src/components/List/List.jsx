import React from 'react';
import PropTypes from 'prop-types';

import './list.scss';

import Item from './Item';
const projects = require('../../projects.json').projects;

class List extends React.PureComponent {

    static propTypes = {
        children: PropTypes.shape({
            projects: PropTypes.array.isRequired,
           count: PropTypes.number.isRequired,
        })
    }

    state = {
        list: [...projects],
        listCount: projects.length

    }

    makeList = (list) => {

        return list.map (project => {

            return <Item key = {project.id} id = {project.id} name = {project.title} />
        });
    }

    render(){
        return (
            <div className = 'projectsList__list'>
                {this.makeList([...this.state.list])}
            </div>
        )
    }
}

export default List;