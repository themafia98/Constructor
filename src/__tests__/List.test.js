import React from 'react';
import {Provider} from 'react-redux';
import { MemoryRouter} from 'react-router-dom';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import Events from 'events';

import List from '../components/List/List';
import store from '../redux/testReduxStore';
const stream = new Events();
const ListComponent = renderer.create(
    <MemoryRouter>
        <Provider store = {store}>
            <List cabinetStream = {stream} list = {{id: 0, title: 'test1', type: "portfolio"}} />
        </Provider>
    </MemoryRouter>
)


test('List.js test ',function(){

    let snapshotList = ListComponent.toJSON();
    expect(snapshotList).toMatchSnapshot();
});