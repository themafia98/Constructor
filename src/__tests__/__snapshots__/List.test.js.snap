// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List.js test  1`] = `
<div
  className="projectsList__list"
>
  <div
    className="item"
    data-id={0}
  >
    <p
      className="ProjectName"
    >
      Test1
    </p>
    <p
      className="ProjectType"
    >
      landing
    </p>
    <div
      className="projectController"
    >
      <input
        className="projectControllerButton_enter"
        onClick={[Function]}
        type="button"
        value="enter build"
      />
      <input
        className="projectControllerButton_delete"
        onClick={[Function]}
        type="button"
        value="delete"
      />
    </div>
  </div>
</div>
`;
